snuba/datasets/plans/single_storage.py

```diff
@@ -80,10 +85,7 @@ def __init__(
         self.__post_processors = post_processors or []
 
     def build_plan(self, request: Request) -> ClickhouseQueryPlan:
-        # TODO: The translator is going to be configured with a mapping between logical
-        # and physical schema that is a property of the relation between dataset (later
-        # entity) and storage.
-        clickhouse_query = QueryTranslator().translate(request.query)
+        clickhouse_query = QueryTranslator(self.__mappers).translate(request.query)
```

Comment:
Not a leading question, but why build the translator here rather than in the constructor or passing it (or them, in the case of multiple storages as noted below) into the query plan builder?

Reply:
because for joins, or for multi table storages, we would build the translator by combining multiple sets of rules (one per storage we are joining), and the multiple sets depend on the query (how we select the storages), so the constructor of the query plan builder does not know the final set of rules to apply till it is time to do the translation.
For single table storages where there is no storage selection, rules never change so technically we could pass in the translator.
For selectable storages (but exclusive) like discover we could pass multiple instances of the translator.

It seems to me that, when looking at all those cases together having a consistent interface for the query plan builder that couples the dataset with rules and not with rules in some cases and a translator in others, would be more consistent. But I think the difference is marginal, so no strong opinion.